src/PerfView/GuiUtilities/StatusBar/StatusBar.xaml.cs

```diff
@@ -203,54 +204,67 @@ public void StartWork(string message, Action work, Action finally_ = null)
             // This part may take a little bit of time, so we pass it off to another
             // thread (to keep the UI responsive, and then when it is done call
             // back (this.BeginInvoke), to finish it off. 
-            ThreadPool.QueueUserWorkItem(delegate
+            ThreadPool.QueueUserWorkItem(delegate (object state)
             {
+                var oldCulture = Tuple.Create(CultureInfo.CurrentCulture, CultureInfo.CurrentUICulture);
+                var culture = (Tuple<CultureInfo, CultureInfo>)state;
                 try
                 {
+                    Thread.CurrentThread.CurrentCulture = culture.Item1;
+                    Thread.CurrentThread.CurrentUICulture = culture.Item2;
+
```

Comment:
My philosophy on culture things is the less the code sees of it the better.    I am completely OK with fixing the ad-hoc parsing that happened to be culture specific.    However when I see explicit manipulation of a thread's culture, I get worried.

Can you describe the rationale here?   Was it for testing?    Ideally very little code 'knows' about the current culture.